src/gui/plot/qgsplotwidget.cpp

```diff
@@ -0,0 +1,834 @@
+    qgsplotwidget.cpp
+    -----------------
+    begin                : August 2025
+    copyright            : (C) 2025 by Mathieu Pellerin
+    email                : mathieu dot opengis dot ch
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsplotwidget.h"
+#include "moc_qgsplotwidget.cpp"
+#include "qgsapplication.h"
+#include "qgsfillsymbol.h"
+#include "qgslinechartplot.h"
+#include "qgslinesymbol.h"
+#include "qgsplotregistry.h"
+#include "qgsbarchartplot.h"
+#include "qgsnumericformatselectorwidget.h"
+
+
+QgsBarChartPlotWidget::QgsBarChartPlotWidget( QWidget *parent )
+  : QgsPlotWidget( parent )
+{
+  setupUi( this );
+
+  setPanelTitle( tr( "Bar Chart Plot Properties" ) );
+
+  mSymbolsList->setColumnCount( 2 );
+  mSymbolsList->setSelectionBehavior( QAbstractItemView::SelectRows );
+  mSymbolsList->setSelectionMode( QAbstractItemView::ExtendedSelection );
+  mSymbolsList->setSortingEnabled( false );
+  mSymbolsList->horizontalHeader()->setSectionResizeMode( 0, QHeaderView::Stretch );
+  mSymbolsList->horizontalHeader()->setSectionResizeMode( 1, QHeaderView::Stretch );
+  mSymbolsList->horizontalHeader()->hide();
+  mSymbolsList->verticalHeader()->hide();
+
+  connect( mAddSymbolPushButton, &QPushButton::clicked, this, &QgsBarChartPlotWidget::mAddSymbolPushButton_clicked );
+  connect( mRemoveSymbolPushButton, &QPushButton::clicked, this, &QgsBarChartPlotWidget::mRemoveSymbolPushButton_clicked );
+
+  mSpinMinXAxis->setClearValue( 0 );
+  connect( mSpinMinXAxis, qOverload<double>( &QDoubleSpinBox::valueChanged ), this, [this]( double ) {
+    if ( mBlockChanges )
+      return;
+    emit widgetChanged();
+  } );
+
+  connect( mSpinMaxXAxis, qOverload<double>( &QDoubleSpinBox::valueChanged ), this, [this]( double ) {
+    if ( mBlockChanges )
+      return;
+    emit widgetChanged();
+  } );
+
+  mSpinMinYAxis->setClearValue( 0 );
+  connect( mSpinMinYAxis, qOverload<double>( &QDoubleSpinBox::valueChanged ), this, [this]( double ) {
+    if ( mBlockChanges )
+      return;
+    emit widgetChanged();
+  } );
+
+  mSpinMaxYAxis->setClearValue( 0 );
+  connect( mSpinMaxYAxis, qOverload<double>( &QDoubleSpinBox::valueChanged ), this, [this]( double ) {
+    if ( mBlockChanges )
+      return;
+    emit widgetChanged();
+  } );
+
+  mXAxisMajorLinesSymbolButton->setSymbolType( Qgis::SymbolType::Line );
```

Comment:
Again, we'll need a registerExpressionContextGenerator method, and to hand that generator off to all the child widgets here.

Reply:
Yep, see above.